components/common/SocialShare.tsx

```diff
@@ -67,49 +89,42 @@ const SocialShare = ({
     }
   };
 
+  const shareContent = encodeURIComponent(
+    `I recommend reading '${title}' by @scriptifed_dev. \n\nRead at - ${url}\n\n`
+  );
+  const facebookShareURL = `https://www.facebook.com/dialog/share?app_id=${process.env.NEXT_PUBLIC_FACEBOOK_APP_ID}&display=popup&href=${url}&redirect_uri=${url}`;
+  const twitterShareURL = `https://twitter.com/intent/tweet?text=${shareContent}`;
+  const whatsAppShareURL = `https://wa.me/?text=${shareContent}`;
+
   return (
     <section className="flex flex-row space-x-4 pt-6">
-      {twitter ? (
-        <a
-          aria-label="Share to Twitter"
-          href={`https://twitter.com/intent/tweet?text=${encodeURIComponent(
-            `I recommend reading '${title}' by ${`@scriptifed_dev`}. \n\nRead at - ${url}\n\n`
-          )}`}
-          className="transition duration-500 ease-in-out transform hover:scale-125"
-          target="_blank"
-          rel="noreferrer"
-        >
-          <TwitterIcon color={`text-${theme}-500`} />
-          {showText && <span>Share to Twitter</span>}
-        </a>
-      ) : null}
-      {whatsApp ? (
-        <a
-          aria-label="Share to Twitter"
-          href={`https://wa.me/?text=${encodeURIComponent(
-            `I recommend reading '${title}' by ${`@scriptifed_dev`}. \n\nRead at - ${url}\n\n`
-          )}`}
-          data-action="share/whatsapp/share"
-          className="transition duration-500 ease-in-out transform hover:scale-125"
-          target="_blank"
-          rel="noreferrer"
-        >
-          <WhatsAppIcon color={`text-${theme}-500`} />
-          {showText && <span>Share to WhatsApp</span>}
-        </a>
-      ) : null}
-      {facebook ? (
-        <a
-          aria-label="Share to Twitter"
-          href={`https://www.facebook.com/dialog/share?app_id=${process.env.NEXT_PUBLIC_FACEBOOK_APP_ID}&display=popup&href=${url}&redirect_uri=${url}`}
-          className="transition duration-500 ease-in-out transform hover:scale-125"
-          target="_blank"
-          rel="noreferrer"
-        >
-          <FacebookIcon color={`text-${theme}-500`} />
-          {showText && <span>Share to Facebook</span>}
-        </a>
-      ) : null}
+      {twitter && (
+        <ShareLink
+          showText={showText}
+          label="Share to Twitter"
+          text="Share to Twitter"
+          url={twitterShareURL}
+          icon={<TwitterIcon color={`text-${theme}-500`} />}
```

Comment:
Also are you guys manually adding these colors to tailwind purge? because dynamically interpolating like this purgecss won't be able to detect them.

Reply:
Oh i see 
![image](https://user-images.githubusercontent.com/35374649/125494062-ddf57470-e926-40ad-8b6c-fe08d5b2dbb8.png)

Reply:
@anuraghazra Yes, we have a [script](https://github.com/scriptified/scriptified/blob/main/scripts/purgableCSSGenerator.js) that extracts the classes that are using the theme variable and then generates purgeable classnames with all possible themes from them. Also, I am not sure if you remember it, but we also had a conversation [regarding this on Twitter](https://twitter.com/psuranas/status/1371064159165579269) 😁, the repo wasn't public back then, so I wasn't able to share the code.

Reply:
Ohhhh hahaha that was a quick round-about of thoughts. yeah I remember. Pretty cool!